apache/yunikorn-core
# [YUNIKORN-2461] Consider supporting /ws/v1/partition/{partitionName}/queue/{queueName}/applications/:state

### What is this PR for?
This PR continues the work from #861.
After contacting @targetoee and receiving his consent, I am proceeding to complete this issue.

After this PR is merged, I will update the API documentation.

### What type of PR is it?


### What is the Jira issue?
[YUNIKORN-2461](https://issues.apache.org/jira/browse/YUNIKORN-2461)

### How should this be tested?

- Ensure all unit tests pass.
  The CI workflow has passed successfully in my [forked repo](https://github.com/blueBlue0102/yunikorn-core/actions/runs/10303868829)
- Successfully fetch the API locally using Postman.

### Screenshots (if appropriate)
![image](https://github.com/user-attachments/assets/786b94f6-35dc-4dbf-890f-571f6915ffce)
![image](https://github.com/user-attachments/assets/59733253-571c-40b0-8347-06092a5f9889)
![image](https://github.com/user-attachments/assets/7e2a7628-9970-4858-a766-5423f5277369)
![image](https://github.com/user-attachments/assets/923c50de-085a-4b4f-89a1-8bcb453cc2a1)
